File: src/Domain/Model/AuthTokens.php
<?php

declare(strict_types=1);

namespace App\Domain\Model;

final class AuthTokens
{
    public function __construct(
        private string $accessToken,
        private int $expiresIn,
        private string $refreshToken,
        private int $refreshExpiresIn,
        private string $tokenType = 'Bearer',
        private ?string $scope = null
    ) {
    }

    public function getAccessToken(): string
    {
        return $this->accessToken;
    }

    public function getExpiresIn(): int
    {
        return $this->expiresIn;
    }

    public function getRefreshToken(): string
    {
        return $this->refreshToken;
    }

    public function getRefreshExpiresIn(): int
    {
        return $this->refreshExpiresIn;
    }

    public function getTokenType(): string
    {
        return $this->tokenType;
    }

    public function getScope(): ?string
    {
        return $this->scope;
    }

    public function toArray(): array
    {
        return [
            'token_type' => $this->tokenType,
            'access_token' => $this->accessToken,
            'expires_in' => $this->expiresIn,
            'refresh_token' => $this->refreshToken,
            'refresh_expires_in' => $this->refreshExpiresIn,
        ];
    }
}

File: src/main/java/br/pucrs/constrsw/oauth/domain/model/AuthTokens.java
package br.pucrs.constrsw.oauth.domain.model;

/**
 * Entidade de dominio: par de tokens emitidos pelo provedor de identidade
 * apos autenticacao com sucesso. Independe de Keycloak/HTTP/JSON.
 */
public final class AuthTokens {

    private final String tokenType;
    private final String accessToken;
    private final Long expiresIn;
    private final String refreshToken;
    private final Long refreshExpiresIn;

    public AuthTokens(String tokenType, String accessToken, Long expiresIn,
                      String refreshToken, Long refreshExpiresIn) {
        this.tokenType = tokenType;
        this.accessToken = accessToken;
        this.expiresIn = expiresIn;
        this.refreshToken = refreshToken;
        this.refreshExpiresIn = refreshExpiresIn;
    }

    public String getTokenType() {
        return tokenType;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public Long getExpiresIn() {
        return expiresIn;
    }

    public String getRefreshToken() {
        return refreshToken;
    }

    public Long getRefreshExpiresIn() {
        return refreshExpiresIn;
    }
}
